tests/test_util.py

```diff
@@ -0,0 +1,134 @@
+import os
+import types
+import unittest
+
+try:
+    from unittest import mock
+except ImportError:
+    # `Python 2` or lower than `Python 3.3` does not
+    # have the `unittest.mock` module built-in
+    import mock
+from pythonforandroid import util
+
+
+class TestUtil(unittest.TestCase):
+    @mock.patch("pythonforandroid.util.makedirs")
+    def test_ensure_dir(self, mock_makedirs):
+        util.ensure_dir("fake_directory")
+        mock_makedirs.assert_called_once_with("fake_directory")
+
+    @mock.patch("shutil.rmtree")
+    @mock.patch("pythonforandroid.util.mkdtemp")
+    def test_temp_directory(self, mock_mkdtemp, mock_shutil_rmtree):
+        mock_mkdtemp.return_value = "/temp/any_directory"
+        with util.temp_directory():
+            mock_mkdtemp.assert_called_once()
+        mock_shutil_rmtree.assert_called_once_with("/temp/any_directory")
+
+    @mock.patch("pythonforandroid.util.chdir")
+    def test_current_directory(self, moch_chdir):
+        chdir_dir = "/temp/any_directory"
+        # test chdir to existing directory
+        with util.current_directory(chdir_dir):
+            moch_chdir.assert_called_once_with("/temp/any_directory")
+        moch_chdir.assert_has_calls(
+            [
+                mock.call("/temp/any_directory"),
+                mock.call(os.getcwd()),
+            ]
+        )
+
+    def test_current_directory_exception(self):
+        # test chdir to non-existing directory, should raise error
+        # for py3 the exception is FileNotFoundError and IOError for py2, to
+        # avoid introduce conditions, we test with a more generic exception
```

Comment:
Minor/question: Just wondering why don't we use docstrings rather than comments like this?
I guess IDE would better understand docstrings for instance and it's usually the way to document a function/method. Why do you think?

Reply:
Yes, I addressed this by documenting the recent modules we wrote,
But be aware that we don't have it linked in our docs source, so it won't be visible until we enable it
:wink:

Reply:
Awesome thanks!